sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -852,6 +842,65 @@ object Radar {
         })
     }
 
+    /**
+     * Tracks the user's location securely with device integrity information.
+     */
+    @JvmStatic
+    fun trackSecure(callback: RadarTrackCallback? = null) {
+        if (!initialized) {
+            callback?.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        val verificationSettings = RadarSettings.getVerificationSettings(context)
+
+        if (!this::fraudManager.isInitialized) {
+            this.fraudManager = RadarFraudManager(this.context, verificationSettings, logger)
+        }
+
+        locationManager.getLocation(RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.HIGH, RadarLocationSource.FOREGROUND_LOCATION, object : RadarLocationCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, stopped: Boolean) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback?.onComplete(status)
+                    }
+
+                    return
+                }
+
+                fraudManager.getIntegrityToken { integrityToken ->
+                    apiClient.track(location, stopped, true, RadarLocationSource.FOREGROUND_LOCATION, false, null, object : RadarApiClient.RadarTrackApiCallback {
```

Comment:
if the function signature changes to 

```
                fraudManager.getIntegrityToken { status, integrityToken ->
```

then we can check that `status` value enum and decide if we should continue to call `apiClient.track` if the integrityToken is available or to return a bad status to `callback`.

See also: `RadarStatus`